src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -145,4 +145,37 @@ public function getDebug()
 
         return $this->debug;
     }
+
+    /**
+     * Returns some or all the existing flash messages:
+     *  * getFlashes() returns all the flash messages
+     *  * getFlashes('notice') returns a simple array with flash messages of that type
+     *  * getFlashes(array('notice', 'error')) returns a nested array of type => messages
+     *
+     * @return array
+     */
+    public function getFlashes($types = null)
+    {
+        // needed to avoid starting the session automatically when looking for flash messages
+        try {
+            $session = $this->getSession();
+            if (null !== $session && !$session->isStarted()) {
+                return array();
+            }
+        } catch (\RuntimeException $e) {
+            return array();
+        }
+
+        if (null === $types || empty($types)) {
+            return $session->getFlashBag()->all();
+        }
+
+        $types = (array) $types;
```

Comment:
I agree that's not very intuitive. Imagine the case where the exact parameter passed to the Twig function is dynamic. You then don't know beforehand whether or not you will get a flat or a nested array. What we could however is to always return the nested array if you pass an array as the argument (no matter the number of arguments) and get the flat result when a string is given.